### PR TITLE
Make container optional

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: tests/Common/InjectorTest.php
Comment (roxblnfk) [inline tests/Common/InjectorTest.php:1]: Please add a test with optional class parameter without container

```php
function (?Foo $foo = null)
```

Comment (vjik) [reply tests/Common/InjectorTest.php:1]: done

**Review by roxblnfk — APPROVED** [review]
It would be nice to move PSR container requirement into `require-dev` section and `suggest`.

**Comment by roxblnfk** [comment]
It would be nice to improve the `MissingRequiredArgumentException` - add a flag whether the Container is passed to the Injector.

And if the Container is not passed, and the argument being resolved is a class, then output a friendly message (`FriendlyException`) will be like "pass an object of the required type as an argument to invoke()/make() or pass a Container when creating the Injector"

If the Container parameter was set, then the friendly message will be like "add an object of the required type to the argument list or make sure that the configured Container can resolve it"

**Pull request merged**